src/pipelines/rust.rs

```diff
@@ -71,13 +99,18 @@ impl RustApp {
         let cargo_features = attrs.get("data-cargo-features").map(|val| val.to_string());
         let keep_debug = attrs.contains_key("data-keep-debug");
         let no_demangle = attrs.contains_key("data-no-demangle");
+        let type_ = attrs
+            .get("data-type")
+            .and_then(|s| RustAppType::from_str(s).ok())
+            .unwrap_or(RustAppType::Main);
```

Comment:
I think we should change this to actually throw an error instead of turning it into an `Option`. Because the way we're doing it here we would just silently take `RustAppType::Main` in case the attribute is missing **or wrong**.

But what we really want is to default to `main` if it's missing and then actually parse it, verifying it's either `main` or `worker` and nothing else.

The following should do it:

```rust
let ty = attrs.get("data-type").as_deref().unwrap_or("main").parse()?;

```

Note: if the compiler can't infer the type you may need to pass it (`.parse::<RustAppType>()?;`).